core/src/main/java/org/apache/accumulo/core/clientImpl/TableOperationsImpl.java

```diff
@@ -2145,4 +2150,27 @@ public void setTabletHostingGoal(String tableName, Range range, TabletHostingGoa
     }
   }
 
+  @Override
+  public Stream<HostingGoalForTablet> getTabletHostingGoal(final String tableName,
+      final Range range) throws TableNotFoundException {
+    EXISTING_TABLE_NAME.validate(tableName);
+
+    final Text scanRangeStart = (range.getStartKey() == null) ? null : range.getStartKey().getRow();
+    TableId tableId = context.getTableId(tableName);
+
+    TabletsMetadata tabletsMetadata =
+        context.getAmple().readTablets().forTable(tableId).overlapping(scanRangeStart, true, null)
+            .fetch(HOSTING_GOAL, PREV_ROW).checkConsistency().build();
+
+    return tabletsMetadata.stream().peek(tm -> {
+      if (scanRangeStart != null && tm.getEndRow() != null
+          && tm.getEndRow().compareTo(scanRangeStart) < 0) {
+        log.debug(">>>> tablet {} is before scan start range: {}", tm.getExtent(), scanRangeStart);
+        throw new RuntimeException("Bug in ample or this code.");
+      }
+    }).takeWhile(tm -> !(tm.getPrevEndRow() != null
+        && range.afterEndKey(new Key(tm.getPrevEndRow()).followingKey(PartialKey.ROW))))
```

Comment:
I think the following is a bit more clear, but thats just my opinion. Either was is correct.

```suggestion
    }).takeWhile(tm -> tm.getPrevEndRow() == null
        || !range.afterEndKey(new Key(tm.getPrevEndRow()).followingKey(PartialKey.ROW)))
```